src/components/shared/WithSize.js

```diff
@@ -1,35 +1,47 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
 
-import React, { PureComponent } from 'react';
+import * as React from 'react';
 import { findDOMNode } from 'react-dom';
+import type { CssPixels } from '../../types/units';
+
+type Sizing = {
+  width: CssPixels,
+  height: CssPixels,
+};
 
 /**
  * Wraps a React component and makes 'width' and 'height' available in the
  * wrapped component's props. These props start out at zero and are updated to
  * the component's DOM node's getBoundingClientRect().width/.height after the
  * component has been mounted. They also get updated when the window is
  * resized.
+ *
  * Note that the props are *not* updated if the size of the element changes
  * for reasons other than a window resize.
- * @param  {class} Wrapped The class that gets wrapped.
- * @return {class}         The resulting Component class.
  */
-export const withSize = Wrapped =>
-  class WithSizeWrapper extends PureComponent {
-    constructor(props) {
+export function withSize(
+  Wrapped: React.ComponentType<*>
+): React.ComponentType<*> {
```

Comment:
I don't think using `*` works as expected. Have you tried adding errors on purpose to see if Flow was catching them ?

You can look at https://flow.org/en/docs/react/hoc/#toc-injecting-props-with-a-higher-order-component for more information about how to type a HOC that injects some props.

You can also have a look at Viewport.js that's more complex as it also consumes some props.

Reply:
Holy cow that was wild to type with generics.